Plugins/Bugsnag/Source/Bugsnag/Private/Apple/ApplePlatformBugsnag.cpp

```diff
@@ -25,9 +25,51 @@
 
 DEFINE_PLATFORM_BUGSNAG(FApplePlatformBugsnag);
 
+#if UE_EDITOR
+
+#import <Bugsnag/BSG_KSCrashReportWriter.h>
+#import <BugsnagPrivate/BugsnagConfiguration+Private.h>
+
+#define IS_PIE_WORLD_KEY "isPlayInEditorWorld"
+#define UNREAL_ENGINE_KEY "unrealEngine"
+
+static void FApplePlatformBugsnag_OnCrashHandler(const BSG_KSCrashReportWriter* writer)
+{
+	writer->beginObject(writer, UNREAL_ENGINE_KEY);
+	writer->addBooleanElement(writer, IS_PIE_WORLD_KEY, GIsPlayInEditorWorld);
+	writer->endContainer(writer);
+}
+
+#endif
+
 void FApplePlatformBugsnag::Start(const TSharedRef<FBugsnagConfiguration>& Configuration)
 {
-	BugsnagClient* Client = [Bugsnag startWithConfiguration:FApplePlatformConfiguration::Configuration(Configuration)];
+	BugsnagConfiguration* CocoaConfig = FApplePlatformConfiguration::Configuration(Configuration);
+#if UE_EDITOR
+	if (GIsEditor || GIsBuildMachine)
```

Comment:
When would `GIsBuildMachine` be set?  By which I mean what is the scenario under which the game would run as a "build machine process"?  Just wondering
a) why it's lumped in with editor and
b) why it's not called out in the log messages

Oh I see elsewhere you don't even start the module if `GIsBuildMachine`.  Presumably that usurps this?

Reply:
It's the Editor that can run as a "build machine process", not the game.

In addition to being a GUI application, the Editor is also launched as a command line tool to do various things, like compile shaders, run automation tests.

Here I'm using `GIsBuildMachine` as an extra flag to indicate that we're not running as the game.

Thinking about it, it would be clearer if we used if `(!FApp::IsGame())` 🤔

Reply:
Removed `GIsBuildMachine` check